kuma/javascript/src/document-provider.jsx

```diff
@@ -157,24 +137,35 @@ export default function DocumentProvider(
 
             let url = link.href;
             let parts = link.pathname.split('/');
+
+            // If the link is not to a /docs/ URL then we can't handle
+            // it via client-side navigation.
             if (parts[2] !== 'docs') {
                 return;
             }
 
+            // If the locale in the URL is not the same as the current
+            // locale then we can't do client-side navigation because
+            // we don't have the right strings loaded.
+            if (parts[1] !== locale) {
```

Comment:
Wait, no I think I'm wrong here. We could still be on a Swahili page (`/sw/docs/...`) and that page theoretically could have a link to another document that doesn't exist in Swahili. I say theoretically because no example of this case may actually exist. I haven't found one yet.